src/users/transform-vc/transform-vc.pipe.ts

```diff
@@ -0,0 +1,12 @@
+import { ArgumentMetadata, Injectable, PipeTransform } from '@nestjs/common';
+import { MessageDto } from '../dto/message.dto';
+
+@Injectable()
+export class TransformVcPipe implements PipeTransform {
+  transform(value: any, metadata: ArgumentMetadata) {
+    if (metadata.type === 'body') {
```

Comment:
unclear what this is doing

Reply:
this is a pipe (data middleware specializes in transforming data ) it transform the body of the request from a vc to a Message object